src/components/loader/loader.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { testStyledSystemMargin } from "../../__spec_helper__/__internal__/test-utils";
+import Loader from ".";
+import useMediaQuery from "../../hooks/useMediaQuery";
+
+jest.mock("../../hooks/useMediaQuery", () => ({
+  __esModule: true,
+  default: jest.fn().mockReturnValue(false),
+}));
+
+testStyledSystemMargin((props) => <Loader {...props} />);
+
+test("when the user disallows animations or their preference cannot be determined, alternative loading text is rendered", () => {
+  render(<Loader />);
+
+  expect(screen.getByText("Loading")).toBeInTheDocument();
+});
+
+describe("when the user allows animations", () => {
+  beforeEach(() => {
+    const mockUseMediaQuery = useMediaQuery as jest.MockedFunction<
+      typeof useMediaQuery
+    >;
+    mockUseMediaQuery.mockReturnValueOnce(true);
+  });
+
+  test("renders three square animation", () => {
+    render(<Loader />);
+
+    const squares = screen.getAllByTestId("loader-square");
+
+    expect(squares).toHaveLength(3);
+  });
+
+  test("root element has accessible name", () => {
+    render(<Loader />);
+
+    expect(screen.getByLabelText("Loading")).toBeInTheDocument();
+  });
+
+  test("when custom `aria-label` is passed, set accessible name to its value", () => {
+    render(<Loader aria-label="Still loading" />);
+
+    expect(screen.getByLabelText("Still loading")).toBeInTheDocument();
+  });
```

Comment:
**suggestion(non-blocking)**: what do you think of using the `toHaveAccessibleName` assertion, so the error messaging aligns more closely to each test's objective?:

```suggestion
    expect(screen.getByRole("progressbar")).toHaveAccessibleName("Loading");
  });

  test("when custom `aria-label` is passed, set accessible name to its value", () => {
    render(<Loader aria-label="Still loading" />);

    expect(screen.getByRole("progressbar")).toHaveAccessibleName("Still loading");
  });
```

Reply:
Yeah, I'm happy to do that. I'll get on it now ✈️